Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +15,214 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
-        }
+            if (IsNullOrEmpty(peopleAndBalances)) return Constants.Nothing;
+            Thread.CurrentThread.CurrentCulture = Config.Culture();
 
+            var (names, money, isPlural) = ReturnNamesMoneyAndIsPlural(peopleAndBalances, "max");
+
+            return $"{names} had the most money ever. {FixMoneyFormat(money)}.";
+        }
+        
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peopleAndBalances)) return Constants.Nothing;
+            Thread.CurrentThread.CurrentCulture = Config.Culture();
+
+            var (names, money, isPlural) = ReturnNamesMoneyAndIsPlural(peopleAndBalances, "min");
+            if (money >= 0) return "N/A.";
+
+            return $"{names} lost the most money. {FixMoneyFormat(money)}.";
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peopleAndBalances)) return Constants.Nothing;
+            Thread.CurrentThread.CurrentCulture = Config.Culture();
+
+            var (names, money, isPlural) = ReturnNamesMoneyAndIsPlural(peopleAndBalances, "rich");
+            var dictionary = new string[]
+            {
+                (isPlural) ? "are" : "is",          // [0]
+                (isPlural) ? "people" : "person"    // [1]
+            };
+
+            return $"{names} {dictionary[0]} the richest {dictionary[1]}. {FixMoneyFormat(money)}.";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsNullOrEmpty(peopleAndBalances)) return Constants.Nothing;
+            Thread.CurrentThread.CurrentCulture = Config.Culture();
+
+            var (names, money, isPlural) = ReturnNamesMoneyAndIsPlural(peopleAndBalances, "poor");
```

Comment:
It's interesting to see object deconstruction. I don't think it's that great, to me this looks like flexing 🍡